### PR TITLE
fs: system: optimize reflink

### PR DESCRIPTION
**Pull request by efiop**
10000 reflinks now take about 1sec instead of 2.7sec

### CODE REVIEW

File: src/dvc_objects/fs/system.py [hunk 1]
```diff
@@ -58,13 +60,20 @@ def _cdll(name):
...
             errno.ENOTSUP,
             "'clonefile' not supported by the standard library",
         )
 
-    clonefile = clib.clonefile
+    return clonefile
+
+
+def _reflink_darwin(src: "AnyFSPath", dst: "AnyFSPath") -> None:
+    import ctypes
```
Comment (skshetry) [inline src/dvc_objects/fs/system.py:74]: Consider moving this outside.

File: src/dvc_objects/fs/system.py [hunk 2]
```diff
@@ -58,13 +60,20 @@ def _cdll(name):
...
+
+    clonefile = _clonefile()
```
[the remark this answers is not in the document]
Comment (efiop) [reply src/dvc_objects/fs/system.py:76]: totally, my oversight

File: src/dvc_objects/fs/system.py [hunk 3]
```diff
@@ -36,7 +37,8 @@ def symlink(source: "AnyFSPath", link_name: "AnyFSPath") -> None:
     os.symlink(source, link_name)
 
 
-def _reflink_darwin(src: "AnyFSPath", dst: "AnyFSPath") -> None:
+@functools.lru_cache(maxsize=1)
+def _clonefile():
```
Comment (skshetry) [inline src/dvc_objects/fs/system.py:41]: Is it worth running a `logger.debug`? Is that expected?

Comment (efiop) [reply src/dvc_objects/fs/system.py:41]: I would rather know that we are falling back there, so yeah. This is ran 1 time worst case, so not much reason to remove.

Comment (skshetry) [reply src/dvc_objects/fs/system.py:41]: yeah, that's true.

**Pull request merged**